cdc/cdc_options.hh

```diff
@@ -47,13 +56,15 @@ public:
     sstring to_sstring() const;
 
     bool enabled() const { return _enabled; }
-    bool preimage() const { return _preimage; }
+    bool preimage() const { return _preimage != image_mode::off; }
+    bool full_preimage() const { return _preimage == image_mode::full; }
     bool postimage() const { return _postimage; }
     delta_mode get_delta_mode() const { return _delta_mode; }
     int ttl() const { return _ttl; }
 
     void enabled(bool b) { _enabled = b; }
-    void preimage(bool b) { _preimage = b; }
+    void preimage(bool b) { preimage(image_mode::on); }
```

Comment:
This looks wrong :-( Maybe `preimage(b ? image_mode::on : image_mode::off)`?

Reply:
Oops. Yes.